crates/bvs-vault-cw20-tokenized/tests/integration_test.rs

```diff
@@ -0,0 +1,1240 @@
+        });
+        vault.execute(app, &staker, &msg).unwrap();
+
+        let balance = cw20.balance(app, &staker);
+        assert_eq!(balance, 10000u128);
+    }
+
+    // queue withdrawal to staker, redeem withdrawal to staker with wrong info.sender
+    {
+        let msg = ExecuteMsg::QueueWithdrawalTo(RecipientAmount {
+            recipient: staker.clone(),
+            amount: Uint128::new(10000),
+        });
+        vault.execute(app, &staker, &msg).unwrap();
+
+        let msg = ExecuteMsg::RedeemWithdrawalTo(Recipient(new_staker.clone()));
+
+        app.update_block(|block| {
+            block.time = block.time.plus_seconds(115);
+        });
+        let err = vault.execute(app, &new_staker, &msg).unwrap_err();
+        assert_eq!(
+            err.root_cause().to_string(),
+            VaultError::Zero {
+                msg: "No queued shares".into()
+            }
+            .to_string()
+        );
+    }
+}
+
+#[test]
+fn test_vault_info() {
+    let app = &mut App::default();
+    let tc = TestContracts::init(app);
+
+    let response: VaultInfoResponse = tc.vault.query(app, &QueryMsg::VaultInfo {}).unwrap();
+    assert_eq!(
+        response,
+        VaultInfoResponse {
+            total_shares: Uint128::new(0),
+            total_assets: Uint128::new(0),
+            router: tc.router.addr,
+            pauser: tc.pauser.addr,
+            operator: app.api().addr_make("operator"),
+            slashing: false,
+            asset_id: format!("cosmos:cosmos-testnet-14002/cw20:{}", tc.cw20.addr).to_string(),
+            contract: "crates.io:bvs-vault-cw20-tokenized".to_string(),
+            version: env!("CARGO_PKG_VERSION").to_string(),
+        }
+    );
+}
+
+#[test]
+fn test_cw20_semi_compliance() {
+    // tokenized vault is only semi compliance
+    // Because we don't allow burning and minting at all except when stake/unstake
+    let app = &mut App::default();
+    let TestContracts {
+        vault,
+        cw20: staking_cw20,
+        ..
+    } = TestContracts::init(app);
+
+    //unlike regular cw20 token, we don't allow external minting
+    // so can't fund it, we'll have to deposit some staking token to have
+    // some amount of receipt token circulating
+    let staker = app.api().addr_make("staker/4545");
+    let msg = ExecuteMsg::DepositFor(RecipientAmount {
+        recipient: staker.clone(),
+        amount: Uint128::new(200),
+    });
+    staking_cw20.increase_allowance(app, &staker, vault.addr(), 100e15 as u128);
+    staking_cw20.fund(app, &staker, 100e15 as u128);
+    vault.execute(app, &staker, &msg).unwrap();
+
+    // can we increase allowance like normal cw20 token would?
+    {
+        let msg = cw20_base::msg::ExecuteMsg::IncreaseAllowance {
+            spender: app.api().addr_make("spender").to_string(),
+            amount: Uint128::new(100),
+            expires: None,
+        };
+        let res = vault.execute(app, &staker, &msg.into());
```

Comment:
```suggestion
        let execute_msg = WasmMsg::Execute {
            contract_addr: vault.addr().to_string(),
            msg: to_json_binary(&msg).unwrap(),
            funds,
        };

        app.execute(staker.clone(), execute_msg.into())
```

So, you don't have to convert `.into()`, you should "call it raw". > The `contract.execute` has generics to make it easier to call.

The underlying, you can just perform this:
https://github.com/satlayer/satlayer-bvs/blob/71430d72abf653ed20a83891e14640aa5598b354/crates/bvs-library/src/testing/contract.rs#L71-L78

Reply:
You can do this for the rest as well.

With this, you don't need `Into`/`From` since the underlying message layer is already in JSON.